pinot-plugins/pinot-stream-ingestion/pinot-kinesis/src/main/java/org/apache/pinot/plugin/stream/kinesis/KinesisConsumer.java

```diff
@@ -194,7 +194,7 @@ private String getShardIterator(String shardId, String sequenceNumber) {
       requestBuilder = requestBuilder.startingSequenceNumber(sequenceNumber)
           .shardIteratorType(ShardIteratorType.AFTER_SEQUENCE_NUMBER);
     } else {
-      requestBuilder = requestBuilder.shardIteratorType(_config.getShardIteratorType());
+      requestBuilder = requestBuilder.shardIteratorType(_config.getInitShardIteratorType());
```

Comment:
Reading through the code again and I'm a little bit confused. In what scenario will we get `null` sequenceNumber?

Reply:
Aah, this PR would make more sense with https://github.com/apache/pinot/pull/13112